### ISSUE TITLE
T-135 WHISPER_FAILED の error_message が whisper のヘルプ全文で、原因を隠す

### ISSUE THREAD

**Issue by shinsuke-terada** [shinsuke-terada/voicedock#135]
## 事実（2026-09-15 実機。#133 の調査中に判明）

**`WHISPER_FAILED` の `error_message` に、whisper-cli のヘルプ全文が入っていた。**

```text
error_code     WHISPER_FAILED
error_message  終了コード 2: rammar to guide decoding
  --grammar-rule RULE               [       ] top-level GBNF grammar rule name
  --grammar-penalty N               [100.0  ] scales down logits of nongrammar tokens
…
```

先頭が `rammar` で始まっているのは、**ヘルプの末尾だけが切り取られている**ためである
（`--grammar GRAMMAR [ ] GBNF g|rammar to guide decoding` の途中で切れている）。

原因は **入力ファイルが存在しないこと**だった。

> **切り詰めそのものは働いていた。**`db.truncate()` により `error_message` は
> ちょうど 200 文字 + `…` で保存されていた。**長さではなく中身の問題である。**whisper-cli は `-f` のファイルを開けないと
**usage を吐いて終了コード 2 で落ちる。**

```text
$ whisper-cli -m …/ggml-large-v3-turbo-q5_0.bin -f /data/staging/4de117f89d5401c6/audio16k.wav
  -vp N,     --vad-speech-pad-ms   N [30     ] VAD speech padding (extend segments)
  -vo N,     --vad-samples-overlap N [0.10   ] VAD samples overlap (seconds between segments)
```

## 何が問題か

**「音声が無い」という一言が出ていれば #133 は即座に分かった。**

実際には `make status` にも `events` にも `WHISPER_FAILED` としか出ず、モデルを疑い、
モデルファイルの存在を確かめ、設定を見直し、ようやく staging が空であることに辿り着いた。
**診断の役に立たないどころか、誤った方向へ誘導した。**

さらに、`error_message` の中身は**ログにもノートにも出うる**。§16 は
`unsafe_log_content: false` で本文を出さないと定めているが、**外部コマンドの
標準エラーをそのまま入れる経路には歯止めが無い。**数 KB のヘルプが DB の 1 カラムに入る。

## いつ起きるか

- **#133 の経路**（`FAILED` の Part の 16 kHz 音声が消される）
- staging を手で消したとき
- ディスク不足で `audio16k.wav` の書き込みが途中で落ちたとき
- **`docker compose down -v` で staging の volume を消したあと、`FAILED` の Part が再評価されたとき**

## 確定した設計判断（**実装者は判断しないこと**）

| # | 判断 |
|---|---|
| 1 | **whisper を起動する前に入力の存在を確かめる。**外部コマンドの終了コードから原因を当てにいかない。**在ることを前提にしている箇所で、前提を確かめる** |
| 2 | **新しいエラーコード `NORMALIZED_MISSING` を §15.1 の「音声」区分へ足す。**`SOURCE_MISSING`（取り込み元が無い）と**意味が違う**。`tests/spec_sync.py` の `spec_error_codes()` が §15.1 の表と `ErrorCode` の一致を見ているので、**両方を同時に直す** |
| 3 | **再正規化を試みる。**inbox に原本が残っていれば（`inbox_retain: raw_saved`）**`NORMALIZING` へ戻して作り直す。**原本も無ければ `NORMALIZED_MISSING` で `FAILED` にする |
| 4 | **`NORMALIZED_MISSING` は §15.2 の再評価対象のままにする。**デバイスを挿し直して墓標を外せば直る余地がある。**終端にしない** |
| 5 | **エラーメッセージに「デバイスから採り直す必要がある」と書く。**利用者が次に何をすればよいかが分かる文言にする（§16.4 の方針） |
| 6 | **切り詰めは既にある。触らない。**`db.truncate()` が 200 文字で切り詰めており、実機の `error_message` も**ちょうど 200 文字 + `…`** だった。長さは問題ではなく、**先頭 200 文字が whisper の usage だった**ことが問題である（起票時の判断 6 / 7 は前提が誤っていた） |
| 7 | **切り詰めの向き（先頭 / 末尾）も変えない。**Python の例外文字列は先頭が本題であり、外部コマンドだけを見て向きを反転させると**他のエラーが読めなくなる。**判断 1 で usage 自体が出なくなるので、この変更は不要である |
## SPEC の改訂

| 箇所 | 内容 |
|---|---|
| §15.1 | `NORMALIZED_MISSING` を「音声」区分に追加（`NORMALIZE_VERIFY_FAILED` の次） |
| §10.6 | 文字起こしの前提として「`normalized_path` が実在すること」を明記。不在時の分岐（再正規化 / `NORMALIZED_MISSING`）を書く |
| §16.4 | 外部コマンドの出力は末尾 500 文字に切り詰めると明記 |

## テスト

| 追加 | 内容 |
|---|---|
| `tests/unit/test_transcribe.py` | 入力不在で **whisper を起動しない**こと（実行ファイルをスタブし、呼ばれたら失敗させる）。`NORMALIZED_MISSING` になること |
| `tests/unit/test_part_resume.py` | inbox に原本があれば `NORMALIZING` へ戻ること。無ければ `NORMALIZED_MISSING` で `FAILED` |
| `tests/unit/test_errors.py` | `truncate_detail()` が 500 文字を超える入力の**末尾**を残すこと。500 以下はそのまま |
| `tests/spec_sync.py` 経由 | §15.1 の表と `ErrorCode` の一致（既存の仕組みが自動で見る） |

### 意図的に壊して確認する（**PR 本文に結果を貼る**）

| 壊す箇所 | 落ちるべきテスト |
|---|---|
| 存在確認を外す | whisper を起動しないテスト |
| §15.1 から `NORMALIZED_MISSING` の行だけ消す | `spec_error_codes()` の整合テスト |
| `truncate_detail()` を先頭 500 文字に変える | 末尾を残すテスト |
| 再正規化の分岐を消す | `NORMALIZING` へ戻るテスト |

**Comment by shinsuke-terada** [comment]
PR #139 で develop へマージ済み。whisper 起動前に `normalized_path` の実在と `size > 0` を確かめ、inbox に原本があれば `NORMALIZING` へ戻し、無ければ `NORMALIZED_MISSING` にする。**必ず `NORMALIZING` を経由する**ため、入力が戻れば §15.2 の再投入が正しい工程へ戻す。SPEC は v5.35（付録 A / 変更 AW-1）。
